### PR TITLE
Independent tasty-core library

### PR DESCRIPTION
**Pull request by bishabosha**

### CODE REVIEW

**Comment by bishabosha** [comment]
I assume we don't actually need bootstrapped/nonbootstrapped variants?

**Comment by smarter** [comment]
If this is to be distributed with the compiler we do actually.

File: project/Build.scala [hunk 1]
```diff
@@ -675,6 +677,7 @@ object Build {
           // library on the compiler classpath since the non-bootstrapped one
           // may not be binary-compatible.
           "dotty-library"       -> packageBin.in(`dotty-library-bootstrapped`, Compile).value,
+          "tasty-core"          -> packageBin.in(`tasty-core-bootstrapped`, Compile).value,
```
Comment (smarter) [inline project/Build.scala]: For the non-bootstrapped compiler this should be "tasty-core" (it's not a library on the compiler classpath so the bootstrapped version isn't needed here), tasty-core-bootstrapped should only be used in the redefinition of packageAll in bootstrapedDottyCompilerSettings.

Comment (bishabosha) [reply project/Build.scala]: Thanks :)

**Comment by bishabosha** [comment]
I can't tell if this is passing due to os-lib failing

**Comment by bishabosha** [comment]
@smarter is this ok to go?

**Comment by bishabosha** [comment]
Actually I would need help cross building this to Scala 2

File: tasty/src/dotty/tools/tasty/TastyFormat.scala [hunk 2]
```diff
@@ -503,7 +503,6 @@ object TastyFormat {
        | DEFAULTparameterized
        | STABLE
        | EXTENSION
-       | GIVEN
```
Comment (smarter) [inline tasty/src/dotty/tools/tasty/TastyFormat.scala]: unrelated change ?

Comment (bishabosha) [reply tasty/src/dotty/tools/tasty/TastyFormat.scala]: Its a duplicated alternative - which is why I made https://github.com/lampepfl/dotty/issues/7643

Comment (smarter) [reply tasty/src/dotty/tools/tasty/TastyFormat.scala]: ok, it's best to put this sort of things in separate commits to make it clear it was intentional.

Comment (bishabosha) [reply tasty/src/dotty/tools/tasty/TastyFormat.scala]: ok I've split into 2 commits

File: compiler/src/dotty/tools/dotc/core/tasty/TreeBuffer.scala [hunk 3]
```diff
@@ -3,8 +3,11 @@ package dotc
 package core
 package tasty
 
-import util.Util.{bestFit, dble}
-import TastyBuffer.{Addr, NoAddr, AddrWidth}
+import dotty.tools.tasty.util.Util.dble
+import dotty.tools.tasty.TastyBuffer
+import TastyBuffer.{ Addr, NoAddr, AddrWidth }
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/core/tasty/TreeBuffer.scala]: Try to avoid changing the import format we use (no spaces around the list of imports)

Comment (bishabosha) [reply compiler/src/dotty/tools/dotc/core/tasty/TreeBuffer.scala]: I've fixed it

**Comment by bishabosha** [comment]
I've noticed that I haven't added this new directory to the BootstrappedOnlyCompilationTests

**Comment by SethTisue** [comment]
@bishabosha is there documentation for this?

**Comment by bishabosha** [comment]
> @bishabosha is there documentation for this?

@SethTisue currently there is only the API docs at http://dotty.epfl.ch/api/dotty/tools/tasty.html, currently its intended as a basis for forming an independent TASTy library (that defines its own algorithm for unpicking, this library only contains the bare minimum, such as comparing TASTy version numbers)
